src/routes/UserRouter.ts

```diff
@@ -16,7 +23,48 @@ import { db } from '../database';
 // });
 /*
 **/
-export class UsersRouter {
+export class UserRouter {
+  /**
+   *
+   * Get Single User
+   * @param req
+   * @param res
+   * @param next
+   */
+  // function that takes a user and returns an encoded token that is created with at subject and timestamp
+  public static tokenForUser(data: any) {
+    const timestamp = new Date().getTime()
+    // const secretToken = jwt.sign({ sub: user.id, iat: timestamp }, secretKey.secret);
+    const secretToken = jwt.encode({sub: data, iat: timestamp}, secretKey.secret)
+    return secretToken;
+  }
+  public static getUser(req: Request, res: Response, next: NextFunction) {
+    // get the user id from the request
+    const userId = parseInt(req.params.id, 10);
+    // test the connection to the database
+    db.connect()
+      .then((obj: pgPromise.IConnected<{}>) => {
+        // console.log("testing connection");
+        // obj.done(); // success, release the connection;
+      })
+      .catch((error) => {
+        // console.log("ERROR:", error.message);
+      });
+    db.any(`SELECT json_build_object(
```

Comment:
@shonore As discussed in the call last Sunday, this returns an array of a single object with the named parameter `json_build_object`. To flatten the structure, I needed to make these changes:
```typescript
db.one(`SELECT json_build_object( // Changed .any (which returns an array) to .one
      'UserId', u."UserId",
      'Email', u."Email",
      'Username', u."Username",
      'LastName', u."LastName",
      'FirstName', u."FirstName",
      'CompanyId', u."CompanyId",
      'AboutMe', u."AboutMe",
      'Role', u."Role",
      'Permissions', p.*) as user // Aliased the json_build_object to user
      FROM public."Users" u
      JOIN public."Permissions" p ON u."Role" = p."Role" WHERE u."UserId" = $1 LIMIT 1`, userId)
      .then(data => data.user) // Extract data.user from the DB response
      .then(user => res.json(user))
      .catch(err => {
        if (err instanceof pgPromise.errors.QueryResultError) { // Added basic error handling
          // The requested user was not found, respond with a 404 Not Found
          res.status(404).end();
        }
      })
      .catch((err) => next(err));
```

Reply:
Sounds good! Did you want to make this change on your branch or did you want me to just update this on my branch?

Reply:
So from our discussion today you said you would add the above code changes in a new pull request on your branch